### PR TITLE
NV 2461 - Delete Tenant Endpoint

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Adds tenant delete use case.

### Why was this change needed?

In order to provide the ability to delete redundant tenants from the platform.

### CODE REVIEW

File: apps/api/src/app/tenant/dtos/delete-tenant.response.dto.ts
Comment (djabarovgeorge) [inline apps/api/src/app/tenant/dtos/delete-tenant.response.dto.ts:1]: Duplicated this one from another class, i wonder if we need to create a more common one for all the deletes or to create a shared interface for those DTOs.

File: apps/api/src/app/tenant/e2e/get-tenant.e2e.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import axios, { AxiosResponse } from 'axios';
 import { UserSession } from '@novu/testing';
 import { TenantRepository } from '@novu/dal';
 
-describe('Get Tenant - /:tenantId (GET)', function () {
+describe('Get Tenant - /tenants/:identifier (GET)', function () {
```
Comment (djabarovgeorge) [inline apps/api/src/app/tenant/e2e/get-tenant.e2e.ts:7]: typo

File: apps/api/src/app/tenant/tenant.controller.ts [hunk 2]
```diff
@@ -83,21 +88,21 @@ export class TenantController {
     );
   }
 
-  @Put('/:tenantId')
+  @Put('/:identifier')
```
Comment (djabarovgeorge) [inline apps/api/src/app/tenant/tenant.controller.ts:91]: I think that 'identifier' would be more intuitive in this context.

File: apps/api/src/app/tenant/tenant.controller.ts [hunk 3]
```diff
@@ -106,4 +111,25 @@ export class TenantController {
...
+
+  @Delete('/:identifier')
+  @ExternalApiAccessible()
+  @UseGuards(JwtAuthGuard)
+  @ApiResponse(DeleteTenantResponseDto)
+  @ApiOperation({
+    summary: 'Delete tenant',
+    description: 'Deletes a tenant entity from the Novu platform',
+  })
+  async removeTenant(
```
Comment (LetItRock) [inline apps/api/src/app/tenant/tenant.controller.ts:123]: Let's be more consistent and make this endpoint similar to what we have in the `topics` controller, I mean the response and the status.

Comment (djabarovgeorge) [reply apps/api/src/app/tenant/tenant.controller.ts:123]: thanks love the suggestion, will add it for sure.

Comment (LetItRock) [reply apps/api/src/app/tenant/tenant.controller.ts:142]: you are still returning the response but the status is "no content" ;)

Comment (djabarovgeorge) [reply apps/api/src/app/tenant/tenant.controller.ts]: actually the `@HttpCode(HttpStatus.NO_CONTENT)` do the heavy lifting and removed the data from the response but there were definitely leftovers on the flow, remove the unnecessary remaining code.

**Review by LetItRock — APPROVED** [review]

**Pull request merged**